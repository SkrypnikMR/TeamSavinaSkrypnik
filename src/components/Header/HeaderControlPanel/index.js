import { withRouter } from 'react-router';
import HeaderControlPanel from './HeaderControlPanel.jsx';
import { connect } from 'react-redux';
import { setValue } from '/src/Store/user/actions';
import { userThemeMode, userNotifSettings } from '/src/Store/user/selectors';
import { logOutAction } from '/src/Store/chat/actions';
import { onlineUsersCount } from '/src/Store/chat/selectors';

const mapStateToProps = state => ({
    onlineUsersCount: onlineUsersCount(state),
    themeMode: userThemeMode(state),
    userNotifSettings: userNotifSettings(state),
});
const mapDispatchToProps = dispatch => ({
    setValue: payload => dispatch(setValue(payload)),
    logOut: () => dispatch(logOutAction()),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderControlPanel));
