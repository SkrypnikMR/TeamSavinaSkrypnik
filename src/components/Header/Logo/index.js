import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Logo from './Logo.jsx';
import { setValue } from '../../../Store/user/actions';

const mapDispatchToProps = dispatch => ({ setValue: payload => dispatch(setValue(payload)) });

export default connect(null, mapDispatchToProps)(withRouter(Logo));
