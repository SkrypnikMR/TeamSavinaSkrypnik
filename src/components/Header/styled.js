import styled from 'styled-components';
import { TRANSPARENT_BACKGROUND } from '/src/components/UI/baseLayout';

export const StHeader = styled.div`
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:10vh;
    ${({ backgroundColor = TRANSPARENT_BACKGROUND }) => backgroundColor && `background-color: ${backgroundColor}`};
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
`;
