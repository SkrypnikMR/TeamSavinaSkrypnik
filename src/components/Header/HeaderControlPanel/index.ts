import { withRouter } from 'react-router';
import HeaderControlPanel from './HeaderControlPanel';

export default withRouter(HeaderControlPanel);
