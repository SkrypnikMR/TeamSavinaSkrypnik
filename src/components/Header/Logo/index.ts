import { withRouter } from 'react-router';

import Logo from './Logo';

export default withRouter(Logo);
