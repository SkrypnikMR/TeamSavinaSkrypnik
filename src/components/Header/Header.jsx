import React from 'react';
import { StHeader } from './styled';
import Logo from './Logo';
import HeaderControlPanel from './HeaderControlPanel';

const Header = () => {
    return (
        <StHeader >
            <Logo />
            <HeaderControlPanel />
        </StHeader >
    );
};

export default Header;
