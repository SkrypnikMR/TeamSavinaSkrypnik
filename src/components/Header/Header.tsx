import React from 'react';

import Logo from './Logo';
import HeaderControlPanel from './HeaderControlPanel';
import { useTheme } from '../Hook/useTheme';

import { StHeader } from './styled';

const Header = () => {
    const { colors, theme } = useTheme();
    return (
        <StHeader colors={colors} theme={theme}>
            <Logo />
            <HeaderControlPanel />
        </StHeader >
    );
};

export default Header;
