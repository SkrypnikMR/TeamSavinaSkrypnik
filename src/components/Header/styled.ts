import styled from 'styled-components';
import { IDiv } from './types';

export const StHeader = styled.div < IDiv > `
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:10vh;
    background: ${({ colors, theme }) => colors[theme].background};
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`}
`;
