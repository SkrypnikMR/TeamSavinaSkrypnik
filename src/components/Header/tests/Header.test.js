import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import Header from '../Header';

const mockStore = configureStore();
const store = mockStore({
  user: {
    themeMode: 'light',
  },
});
describe('Header', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Header />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('Logo')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('HeaderControlPanel')).toHaveLength(1);
    });
});
