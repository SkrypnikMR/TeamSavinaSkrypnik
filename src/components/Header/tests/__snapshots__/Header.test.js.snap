// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<div class=\\"styled__StHeader-zzrvsn-0 hDaupM\\"><div class=\\"styled__StLogo-sc-9vy3sh-0 dOfUuH\\"><img src=\\"./public/assets/images/logo.png\\"/><h1>logo</h1></div><div class=\\"styled__StControl-sc-1i99xqk-0 joBAxQ\\"><button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">ðŸŒ—</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">english_translate_btn</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">russian_translate_btn</button><button id=\\"ar\\" type=\\"button\\" value=\\"ar\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">arabian_translate_btn</button></div></div>"`;
